network/proto/network.proto

```diff
@@ -6,32 +6,46 @@ import "github.com/micro/go-micro/router/proto/router.proto";
 
 // Network service is usesd to gain visibility into networks
 service Network {
-        rpc ListRoutes(go.micro.router.Request) returns (go.micro.router.ListResponse) {};
         rpc ListNodes(ListRequest) returns (ListResponse) {};
-        rpc Neighbourhood(NeighbourhoodRequest) returns (NeighbourhoodResponse) {};
+        rpc ListPeers(PeerRequest) returns (PeerResponse) {};
```

Comment:
I'm assuming Neighbourhood is being replaced by Topology. And now we also have ListPeers which seems to make ListNodes redundant.

Maybe we should remove ListNodes and just have

Network
- Peers
- Routes
- Topology

Reply:
Yes, `Topology()` does replace `Neighbourhood()`

We can remove `ListNodes`, of course. Probably makes a lot of sense.

One thing I'm not sure now regarding your comment, though, is: are you suggesting `Topology()` should return the **whole known graph** as opposed to the originally intended graph up to `MaxDepth=3`?

Reply:
Bad wording. Yes just max depth 3. We're limiting to the depth to 3 on processing peer messages so the "whole known graph" is basically that.

Reply:
Below you mentioned you want to return topology (known graph up to 3 hops) in `Peers()` in `network.Network` and chuck `Nodes()/Topology()`. Does that mean we eliminate `handler.Topoly`, too? I'm assuming it does?

Reply:
That original logic was broken and inefficient as it was working around the inaccessibility of the full graph it was doing this:
* [get the whole graph i.e. all nodes](https://github.com/micro/go-micro/blob/master/network/handler/handler.go#L73)
* [search through all nodes](https://github.com/micro/go-micro/blob/master/network/handler/handler.go#L78)

That's pretty bad and the only reason for at the time was the lack of "access" to the network graph; the only access to the network graph we had was via the `Nodes()` method of the `network.Node` which was returning the whole graph we could do some processing on. Hence, we traversed the whole graph, collected all the nodes and built subgraph by searching.

In other words: the original solution you keep mentioning was really just a "placeholder" workaround until we would figure out proper node/network interface. Now, if we scrap `Nodes()`, too I'm wondering how to build up the graph etc...that's the point I was trying to explain to you yesterday.